From c39bb3a5c69206d8917cd0089a1b6c4bae29527f Mon Sep 17 00:00:00 2001
From: Levente Kurusa <levex@linux.com>
Date: Sat, 14 Jan 2017 21:27:18 +0000
Subject: [PATCH 2/6] macOS: qemu: silence raw format warning

Signed-off-by: Levente Kurusa <levex@linux.com>
---
 src/utils/pintos | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/utils/pintos b/src/utils/pintos
index 7ece59a..a7f2b26 100755
--- a/src/utils/pintos
+++ b/src/utils/pintos
@@ -619,10 +619,10 @@ sub run_qemu {
     print "warning: qemu doesn't support jitter\n"
       if defined $jitter;
     my (@cmd) = ('qemu-system-x86_64');
-    push (@cmd, '-hda', $disks[0]) if defined $disks[0];
-    push (@cmd, '-hdb', $disks[1]) if defined $disks[1];
-    push (@cmd, '-hdc', $disks[2]) if defined $disks[2];
-    push (@cmd, '-hdd', $disks[3]) if defined $disks[3];
+    push (@cmd, '-drive', 'format=raw,media=disk,index=0,file=' . $disks[0]) if defined $disks[0];
+    push (@cmd, '-drive', 'format=raw,media=disk,index=1,file=' . $disks[1]) if defined $disks[1];
+    push (@cmd, '-drive', 'format=raw,media=disk,index=2,file=' . $disks[2]) if defined $disks[2];
+    push (@cmd, '-drive', 'format=raw,media=disk,index=3,file=' . $disks[3]) if defined $disks[3];
     push (@cmd, '-m', $mem);
     push (@cmd, '-net', 'none');
     push (@cmd, '-nographic') if $vga eq 'none';
-- 
2.10.2

